src/js/media-chrome-dialog.ts

```diff
@@ -94,19 +96,26 @@ class MediaChromeDialog extends globalThis.HTMLElement {
   constructor() {
     super();
 
+    this.addEventListener('invoke', this);
+    this.addEventListener('focusout', this);
+    this.addEventListener('keydown', this);
+  }
+
+  #init() {
+    if (this.#isInit) return;
+    this.#isInit = true;
+
     if (!this.shadowRoot) {
       // Set up the Shadow DOM if not using Declarative Shadow DOM.
       this.attachShadow({ mode: 'open' });
 
-      this.nativeEl = (
-        this.constructor as typeof MediaChromeDialog
-      ).template.content.cloneNode(true) as HTMLElement;
-      this.shadowRoot.append(this.nativeEl);
-    }
+      const attrs = namedNodeMapToObject(this.attributes);
 
-    this.addEventListener('invoke', this);
-    this.addEventListener('focusout', this);
-    this.addEventListener('keydown', this);
+      this.shadowRoot.innerHTML = /*html*/ `
```

Comment:
question: this is a pretty different implementation/architecture from other "base" components (e.g. the relationship between `media-chrome-button` vs. `media-play-button`, where the `slotTemplate` is provided via the subclass's `constructor`. Any reason(s) for this different implementation approach?

Reply:
yes, the reason is that passing args through the constructor is not really a good practice. I might have mentioned this before. basically if you create a DOM element there is no possibility to pass arguments. `document.createElement('my-button')`. I know it's a base class and it's possible with the constructor like we do but still this is more of an unexpected API than using a static method and overriding static methods in the child class. I've been wanting to change those base classes for a while and this new element seemed like a good candidate to start this process.

Reply:
So would your recommendation longer term be to e.g. refactor our button class defs to more resemble this architecture?